frontend/src/metabase/common/components/DashboardPicker/components/DashboardPicker.tsx

```diff
@@ -197,6 +200,14 @@ const DashboardPickerInner = (
     [currentCollection, userPersonalCollectionId, onPathChange],
   );
 
+  useEnsureCollectionSelected({
+    currentCollection,
+    enabled: path === defaultPath,
+    options,
+    useRootCollection: initialValue?.id == null,
```

Comment:
Is it okay for Our analytics? Can we check only `initialValue == null` or something like that?

Reply:
> Is it okay for Our analytics? 

Seems to be working fine.

> Can we check only `initialValue == null` or something like that?

I came across a case where `initialValue` had `id: null` and it didn't work correctly.